src/main/java/io/github/mzmine/gui/mainwindow/MainWindowController.java

```diff
@@ -343,4 +343,25 @@ public void setSelectedTab(String tabName) {
         }
     }
 
+    public void updateTabName(String tabName) {

```

Comment:
Referencing the tab using a String is kinda ugly in Java..
Why don't you add fx:id to the <Tab> in the main FXML file, and reference it in the controller using @FXML?

````
@FXML
private Tab rawDataFilesTab, featureTablesTab;
````
Then you can make a simple updateTabName() method in which you call
````
rawDataFilesTab.setText(...);
featureTablesTab.setText(...);
````